### PR TITLE
add example rquickjs-cli

### PR DESCRIPTION
**Pull request by ahaoboy**
This is a very simple example, but it covers many common use cases.
1. call rust functions in js
2. call js functions in rust
3. run js code in rust

### CODE REVIEW

**Comment by trevyn** [comment]
I found this helpful for quickly testing rquickjs, thanks. :)

File: examples/rquickjs-cli/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+"#,
+        )?;
+
+        let console: Object = global.get("console")?;
+        let js_log: Function = console.get("log")?;
+        loop {
+            let mut input = String::new();
+            let _ = stdout().write(b"> ")?;
+            stdout().flush()?;
```
Comment (DelSkayn) [inline examples/rquickjs-cli/src/main.rs:34]: Why manually lock stdout? `print!("> ")` should be sufficient here.

File: examples/rquickjs-cli/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            std::io::stdin().read_line(&mut input)?;
+            let ret = ctx.eval::<Value, _>(input.as_bytes())?;
```
Comment (DelSkayn) [inline examples/rquickjs-cli/src/main.rs:36]: This won't report errors properly. You should retrieve the error thrown by QuickJS by calling for example `.catch(&ctx)` on the returned result.

File: examples/rquickjs-cli/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            js_log.call::<(Value<'_>,), ()>((ret,))?;
```
Comment (DelSkayn) [inline examples/rquickjs-cli/src/main.rs:37]: Same here.

This probably won't throw an error but for an example code it is good to follow best practice.

**Review by DelSkayn — COMMENTED** [review]
I am fine with adding this example, just have some comments.

File: examples/rquickjs-cli/src/main.rs (hunk 1 above)
Comment (ahaoboy) [reply examples/rquickjs-cli/src/main.rs]: If flush is not used here to force the output of the buffer content, the prompt will not be display.

File: examples/rquickjs-cli/src/main.rs (hunk 2 above)
Comment (ahaoboy) [reply examples/rquickjs-cli/src/main.rs]: Is this the right way to use it?
```rust
  match ctx.eval::<Value, _>(input.as_bytes()).catch(&ctx) {
      Ok(ret) => {
          js_log.call::<(Value<'_>,), ()>((ret,))?;
      }
      Err(err) => {
          println!("{err}");
      }
  }
```

image

Comment (DelSkayn) [reply examples/rquickjs-cli/src/main.rs]: Yes that will return the right error.

**Review by DelSkayn — APPROVED** [review]

**Pull request merged**